xarray/core/pycompat.py

```diff
@@ -35,3 +35,10 @@ def is_duck_dask_array(x):
     cupy_array_type = (cupy.ndarray,)
 except ImportError:  # pragma: no cover
     cupy_array_type = ()
+
+try:
+    import pint
+
+    pint_array_type = (pint.Quantity,)
+except ImportError:  # pragma: no cover
+    pint_array_type = ()
```

Comment:
it seems very strange to me that this would make some of the tests fail. `pint` didn't release recently, so this should not happen. I tried to rerun, let's see if that changes anything.

Reply:
this seems to be due to a circular import: `pint.compat` tries to import `xarray` objects. `pytest.importorskip` silences the import error somehow, so we get partially initialized modules. I'm somewhat surprised `pint` still does that, I thought we switched to comparing names some time ago.

Reply:
It was actually the other way around! We had to change from name-based comparison to actual type checks since both xarray and measurements have a `Variable` class, and they need to be treated completely differently (see https://github.com/hgrecco/pint/pull/959).

Reply:
oh, I see. There should be a way to compare without importing (maybe using fully-qualified names like `xarray.Variable` or `xarray.core.variable.Variable`), but I guess to really fix this we need figure out #3950.